ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -728,6 +761,29 @@ void setChunkException(RouterException exception) {
     }
 
     /**
+     * Slice this chunk's data to only include the bytes within the operation's specified byte range.
+     * @param blobData the {@link BlobData} for this chunk.
+     * @return A {@link ByteBuffer} that only includes bytes within the operation's specified byte range.
+     */
+    ByteBuffer filterChunkToRange(BlobData blobData) {
+      ByteBuffer buf = blobData.getStream().getByteBuffer();
+      if (!hasRange) {
+        return buf;
+      }
+      // the (inclusive) byte offset of the beginning of the chunk.
+      long chunkStartOffset = (long) chunkSize * (chunkIndex + firstChunkIndexInRange);
```

Comment:
Yes, the calculation above is a lot cleaner. I realized that I didn't need the confusing `firstChunkIndexInRange` with your calculation.  We know that a chunk is the first chunk when `chunkIndex == 0` and that is the end chunk when `chunkIndex == (numChunksTotal - 1)`